src/Tasks/AssemblyDependency/ReferenceTable.cs

```diff
@@ -1721,33 +1741,42 @@ private bool FindAssociatedFiles()
                         if (!hasFrameworkPath && !reference.EmbedInteropTypes && reference.SDKName.Length == 0)
                         {
-                            // Look for companion files like pdbs and xmls that ride along with 
-                            // assemblies.
-                            if (_findRelatedFiles)
+                            if (!reference.ExternallyResolved)
                             {
-                                FindRelatedFiles(reference);
-                            }
+                                // Look for companion files like pdbs and xmls that ride along with 
+                                // assemblies.
+                                if (_findRelatedFiles)
+                                {
+                                    FindRelatedFiles(reference);
+                                }
 
-                            // Satellite assemblies are named <CultureDir>\<AppBaseName>.resources.dll
-                            // where <CultureDir> is like 'en', 'fr', etc.
-                            if (_findSatellites)
-                            {
-                                FindSatellites(reference);
+                                // Satellite assemblies are named <CultureDir>\<AppBaseName>.resources.dll
+                                // where <CultureDir> is like 'en', 'fr', etc.
+                                if (_findSatellites)
+                                {
+                                    FindSatellites(reference);
+                                }
+
+                                // Look for serialization assemblies.
+                                if (_findSerializationAssemblies)
+                                {
+                                    FindSerializationAssemblies(reference);
+                                }
                             }
 
-                            // Look for serialization assemblies.
-                            if (_findSerializationAssemblies)
+                            if (!reference.ExternallyResolved || FindDependenciesOfExternallyResolvedReferences)
                             {
-                                FindSerializationAssemblies(reference);
+                                // Look for dependent assemblies.
+                                if (_findDependencies)
+                                {
+                                    FindDependenciesAndScatterFiles(reference, newEntries);
+                                }
                             }
-
-                            // Look for dependent assemblies.
-                            if (_findDependencies)
+                            else
                             {
-                                FindDependenciesAndScatterFiles(reference, newEntries);
+                                SkippedFindingExternallyResolvedDependencies = true;
```

Comment:
ComputeClosure clears out _references. This flag probably needs to be reset as well right?

Reply:
Done.